core/src/main/java/io/grpc/internal/BinaryLog.java

```diff
@@ -64,14 +67,31 @@ public String toString() {
         + "maxMessageBytes=" + maxMessageBytes + "]";
   }
 
-  private static final Factory DEFAULT_FACTORY =
-      new Factory(System.getenv("GRPC_BINARY_LOG_CONFIG"));
+  @Nullable
+  private static final Factory DEFAULT_FACTORY;
+
+  static {
+    Factory defaultFactory = null;
```

Comment:
Assign NOOP_LOG instead of `null`? (Also would need changing in the Throwable). Would remove the `if` from `getLog()` without introducing any complexity.